instrument/tracing.go

```diff
@@ -0,0 +1,54 @@
+package instrument
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber-go/tally"
+	jaegercfg "github.com/uber/jaeger-client-go/config"
+	jaegerzap "github.com/uber/jaeger-client-go/log/zap"
+	jaegertally "github.com/uber/jaeger-lib/metrics/tally"
+	"go.uber.org/zap"
+)
+
+// Supported tracing backends. Currently only jaeger is supported.
+var (
+	TracingBackendJaeger = "jaeger"
+)
+
+// TracingConfiguration configures an opentracing backend for m3query to use. Currently only jaeger is supported.
+// Tracing is disabled if no backend is specified.
+type TracingConfiguration struct {
+	Backend string                  `yaml:"backend"`
+	Jaeger  jaegercfg.Configuration `yaml:"jaeger"`
+}
+
+
+func (cfg *TracingConfiguration) NewTracer(serviceName string, scope tally.Scope, logger *zap.Logger) (opentracing.Tracer, io.Closer, error) {
+	if cfg.Backend == "" {
+		return opentracing.NoopTracer{}, noopCloser{}, nil
+	}
+
+	if cfg.Backend != TracingBackendJaeger {
+		return nil, nil, fmt.Errorf("unknown tracing backend: %s", cfg.Backend)
+	}
+
+	cfg.Jaeger.ServiceName = serviceName
```

Comment:
Check to make sure `cfg.Jaeger.ServiceName != ""`

Reply:
Good call.